// Copyright 2023 Intrinsic Innovation LLC
// Intrinsic Proprietary and Confidential
// Provided subject to written agreement between the parties.

syntax = "proto3";

package intrinsic_proto.skills;

import "google/protobuf/any.proto";
import "intrinsic/executive/proto/behavior_tree.proto";
import "intrinsic/skills/proto/skills.proto";

message SkillRegistration {
  intrinsic_proto.skills.SkillHandle validate_handle = 1;
  intrinsic_proto.skills.SkillHandle project_handle = 2;
  intrinsic_proto.skills.SkillHandle execute_handle = 3;
  intrinsic_proto.skills.SkillHandle skill_info_handle = 5;

  intrinsic_proto.skills.Skill skill = 4;
}

message BehaviorTreeRegistration {
  intrinsic_proto.executive.BehaviorTree behavior_tree = 1;
}

message SkillRegistryConfig {
  repeated SkillRegistration skills = 2;
  repeated BehaviorTreeRegistration behavior_trees = 3;
}

message SkillRegistrationOverride {
  google.protobuf.Any default_parameters = 2;
  reserved "default_values";
  reserved 1;
}

message SkillRegistryConfigOverride {
  // Mapping from skill name to registration overrides.
  map<string, SkillRegistrationOverride> overrides = 1;
}
